hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/types/ReplaceUnknownOperandTypeInference.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.types;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.type.SqlOperandTypeInference;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+/**
+ * Type inference, that replaces unknown operands with the given types.
+ */
+public class ReplaceUnknownOperandTypeInference implements SqlOperandTypeInference {
```

Comment:
Missing test in HazelcastInferTypesTest.

Reply:
Created a ticket for that: https://github.com/hazelcast/hazelcast/issues/17360
There is a high level of confidence that it works as expected because there are integration tests that cover this